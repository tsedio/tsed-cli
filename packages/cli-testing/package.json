{
  "name": "@tsed/cli-testing",
  "description": "Utils to test you CLI based on Ts.ED CLI",
  "version": "6.4.4",
  "type": "module",
  "main": "./lib/esm/index.js",
  "source": "./src/index.ts",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "cli-tsed-source": "./src/index.ts",
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "default": "./lib/esm/index.js"
    }
  },
  "scripts": {
    "build": "yarn build:ts",
    "build:ts": "tsc --build tsconfig.esm.json"
  },
  "keywords": [
    "Ts.ED",
    "cli",
    "typescript",
    "framework",
    "rest",
    "api",
    "express",
    "decorators"
  ],
  "dependencies": {
    "@tsed/cli-core": "workspace:*",
    "tslib": "2.7.0"
  },
  "devDependencies": {
    "@tsed/typescript": "workspace:*",
    "cross-env": "7.0.3",
    "typescript": "5.6.2",
    "vitest": "3.2.4"
  }
}
