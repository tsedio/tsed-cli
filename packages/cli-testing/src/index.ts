export * from "./CliPlatformTest.js";
export * from "./FakeCliExeca.js";
export * from "./FakeCliFs.js";
export * from "./FakeCliHttpClient.js";
export * from "./normalizePath.js";
