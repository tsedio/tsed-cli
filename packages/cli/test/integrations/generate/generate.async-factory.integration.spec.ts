// @ts-ignore
import {CliPlatformTest, FakeCliFs} from "@tsed/cli-testing";

import {GenerateCmd} from "../../../src/index.js";

describe("Generate AsyncFactory", () => {
  beforeEach(() =>
    CliPlatformTest.bootstrap({
      commands: [GenerateCmd]
    })
  );
  afterEach(() => CliPlatformTest.reset());

  it("should generate a template with the right options", async () => {
    await CliPlatformTest.initProject();

    await CliPlatformTest.exec("generate", {
      rootDir: "./project-data",
      type: "async.factory",
      name: "TestFactory"
    });

    expect(FakeCliFs.getKeys()).toMatchInlineSnapshot(`
      [
        "project-name",
        "project-name/.barrels.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/.swcrc",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/nodemon.json",
        "project-name/package.json",
        "project-name/processes.config.cjs",
        "project-name/src/Server.ts",
        "project-name/src/config/config.ts",
        "project-name/src/config/logger/index.ts",
        "project-name/src/config/utils/index.ts",
        "project-name/src/controllers/pages/IndexController.ts",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/index.ts",
        "project-name/src/services/TestFactory.ts",
        "project-name/tsconfig.base.json",
        "project-name/tsconfig.json",
        "project-name/tsconfig.node.json",
      ]
    `);

    const result = FakeCliFs.files.get("project-name/src/services/TestFactory.ts");

    expect(result).toMatchInlineSnapshot(`
      "import {injectable} from "@tsed/di";

      interface TestFactoryOptions {
        
      }
          
      declare global {
        namespace TsED {
          interface Configuration extends Record<string, any> {
            testFactory: Options;
          }
        }
      }    

      export const TestFactory = injectable(Symbol.for("TestFactory"))
        .factory(async () => {
          const myConstant = constant<TestFactoryOptions>("testFactory");
          
          // do something async
          await Promise.resolve();
          
          return {};
        })
        .token();
        
      export type {{symbolName}} = typeof TestFactory;"
    `);
  });
});
