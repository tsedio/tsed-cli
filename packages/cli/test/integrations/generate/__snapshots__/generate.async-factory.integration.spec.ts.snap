// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generate AsyncFactory > should generate a template with the right options 1`] = `
"import {Configuration, Inject, registerProvider} from "@tsed/di";

export type Test = any; // implement type or interface for type checking
export const Test = injectable(Symbol.for("Test")
  .asyncFactory(async () => {
    // retrieve custom configuration from settings (e.g.: \`@Configuration({hello: "value"})\`)
    const hello = constant("hello");

    return {hello};
  })
  .token();
"
`;
